webpack/components/NewTemplateSync/components/NewTemplateSyncForm/NewTemplateSyncForm.js

```diff
@@ -0,0 +1,145 @@
+import React from 'react';
+import { change } from 'redux-form';
+import PropTypes from 'prop-types';
+
+import Form from 'foremanReact/components/common/forms/Form';
+
+import SyncSettingsFields from '../SyncSettingFields';
+import SyncTypeRadios from '../SyncTypeRadios';
+import { formName } from './NewTemplateSyncFormConstants';
+
+const submit = syncType => (formValues, dispatch, props) => {
+  const { submitForm, importUrl, exportUrl, history, currentFields } = props;
+  const url = syncType === 'import' ? importUrl : exportUrl;
+  const currentFieldNames = Object.keys(currentFields);
+  const postValues = Object.keys(formValues).reduce((memo, key) => {
+    if (currentFieldNames.includes(key)) {
+      memo[key] = formValues[key];
+    }
+    return memo;
+  }, {});
+
+  return submitForm({
+    url,
+    values: postValues,
+    message: `Templates were ${syncType}ed.`,
+    item: 'TemplateSync',
+  }).then(args => {
+    history.replace({ pathname: '/template_syncs/result' });
+  });
+};
+
+const redirectToResult = history => () =>
+  history.push({ pathname: '/template_syncs/result' });
+
+class NewTemplateSyncForm extends React.Component {
+  allowedSyncType = (userPermissions, radioAttrs) =>
+    this.props.userPermissions[radioAttrs.permission];
+
+  constructor(props) {
+    super(props);
+
+    this.radioButtons = [
+      { label: 'Import', value: 'import', permission: 'import' },
+      { label: 'Export', value: 'export', permission: 'export' },
+    ];
+
+    this.state = {
```

Comment:
Is there a way you can implement it without an internal state?
We have found out that where we mix redux and internal state things get messy and difficult to maintain.

Reply:
Yes, there is a way. I even had that initially but it looked too complex for the task so I decided to just put it in state. I expect this will go away once we move to new forms framework.